NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,228 @@
+//
+//  ScrollableSegmentedControl.swift
+//  ScrollableSegmentedControl
+//
+//  Created by MArko Satlan on 22/07/2018.
+//  Copyright © 2018 MArko Satlan. All rights reserved.
+//
+
+import UIKit
+
+class ScrollableSegmentedControl: UIControl {
+
+// MARK: - Properties
+    // Open
+    open var font = UIFont(name: "Georgia", size: 15) {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonSpacing: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonMargin: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var selectedSegmentIndex: Int = -1 {
+        didSet {
+            highlightSelectedSegment()
+            calculateAndSetScrollOffset()
+      
+            // deselect previous button if different button is tapped
+            if oldValue != selectedSegmentIndex {
+                let button = buttonsArray.filter{$0.tag == oldValue}.first
+                if let previousButton = button {
+                    previousButton.isSelected = false
+                    UIButton.animate(withDuration: 0.3, animations: {
+                        previousButton.transform = CGAffineTransform.identity
+                    })
+                }
+            }
+        }
+    }
+    
+    // Private
+    private var buttonTitles: [String] = []
+    private var buttonsArray: [UIButton] = []
+    private let scrollView = UIScrollView()
+    
+// MARK: - Methods
+    // Init
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        setUp()
+        drawSeparatorLine()
+    }
+    
+    public init(items: [String]) {
+        self.buttonTitles = items
+        
+        super.init(frame: CGRect.zero)
+        
+        setUp()
+        createButtons()
+        drawSeparatorLine()
+    }
+    
+    public func setItems(_ items: [String]) {
+        self.buttonTitles = items
+        createButtons()
+        setNeedsLayout()
+    }
+    
+    // Layout subviews - set frames
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        scrollView.frame = bounds
+        
+        updateUI()
+        highlightSelectedSegment()
+        calculateAndSetScrollOffset()
+    }
+    
+    // Action methods
+    @objc private func buttonTapped(_ sender: UIButton) {
+        let button = buttonsArray.filter{$0.tag == selectedSegmentIndex}.first
+        
+        selectedSegmentIndex = sender.tag
+        
+        if let previousButton = button {
```

Comment:
Similar to the comment above, can be combined into one line. Just be careful because you are setting `selectedSegmentIndex` to a new value in between. It would be more readable if the line `93` was written after dealing with previous button. Currently it "interrupts" the mind flow :)